src/core/exceptions.py

```diff
@@ -104,7 +93,7 @@ def __init__(self, message=""):
 class InvalidNetwork(Exception):
     """Raised when an invalid network is requested."""
 
-    def __init__(self, message=""):
+    def __init__(self, message: str = ""):
         _message = f"""
                 Not supported network: {message}
```

Comment:
Just a minor english thing. Any time it's "Not supported" you can just use "Unsupported"